flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -968,6 +982,75 @@ public void testExceptionHistoryWithRestartableFailure() {
                 lessThanOrEqualTo(updateStateTriggeringJobFailureTimeframe.upperEndpoint()));
     }
 
+    @Test(expected = IllegalStateException.class)
```

Comment:
Shouldn't we do a check for the error message similarly to what is done in `AdaptiveSchedulerTest` here as well as `IllegalStateException` might be quite generic.

Reply:
I guess for symmetric reason we should change it. The thing with testing for exception messages is always that it couples tests quite tightly with internal details.